### ISSUE TITLE
Upgrading to 3.0

### ISSUE THREAD

**Issue by chihoko** [dj-wasabi/puppet-zabbix#166]
v3.0 has been released today, is the module compatible?

**Comment by slashr00t** [comment]
No, it's not. It looks for wrong sql files and has 3 parameters in config template that are invalid for version 3.

**Comment by slashr00t** [comment]
I've got the PR ready - #167

**Comment by dj-wasabi** [comment]
Hi @chihoko 

No, I don't think the module isn't ready yet. I'll start with investigating what is changed and start with it. At the moment, the PR that is provided does not meet the standards.

**Comment by dj-wasabi** [comment]
The PR from @slashr00t is accepted.
But there are some other configuration options for encrypting traffic between Server and Agent that aren't included in this PR, so these will come soon.

**Comment by ericsysmin** [comment]
It needs a lot of fixes, including node declarations to be removed, i will help make it work just give me time

**Comment by ericsysmin** [comment]
Created Pull Request 169 to help address some of the issues so far, with proxy, server, and agent tls support, still need to check for non-accepted values in 3.0 to clean configuration

**Comment by ericsysmin** [comment]
Support is now added for 3.0 zabbix-server, will be working on zabbix-agent this week as well @chihoko

**Comment by chihoko** [comment]
@ericsysmin Cheers, much appreciated!

**Comment by ericsysmin** [comment]
@chihoko if you'd like to test the 3.0 support from my fork you can, I do not place any guarantees on it though and it may not work 100% but I'd love someone to test it and all the functionality. My fork is waiting on a pull request as it needs to be heavily tested for all the edits.

**Comment by chihoko** [comment]
@ericsysmin We've successfully updated to zabbix server v3.0 using your fork. The only issue we had is that you can't specify different versions for zabbix::agent and zabbix::server as zabbix::agent declares the zabbix::repo class first. So we applied v3 to the zabbix agent and server on the Zabbix server only. Once you have finished the zabbix::agent we'll push that out to all our agents. Thanks!

**Comment by ericsysmin** [comment]
oh, I will take a look and see what I can do.

**Comment by ericsysmin** [comment]
Hmmm, that's just an existing problem as well. I can only think the way to work around that is the zabbix::zabbix_version: parameter, or maybe zabbix::params otherwise you need to specify both zabbix::agent::zabbix_version  and zabbix::server::zabbix_version @chihoko

**Comment by chihoko** [comment]
My point was that if you specify both zabbix::agent::zabbix_version and zabbix::server::zabbix_version, the zabbix::repo class will be passed zabbix_version by whichever class declares it first. So you can't have different values for each which makes sense as zabbix doesn't provide different repo's for each.

**Comment by ericsysmin** [comment]
i won't fix that, yum won't allow that to even work, even if you create two seperate repositories, it will always see what is highest version in each, let alone zabbix-proxy 2.4 can't talk to zabbix-server 3.0. 

There's essentially no way to fix it outside of ignorig zabbix::repo altogether and setting manage repo to false, and manually installing the rpms via rpm -ivh <url>
@chihoko

**Comment by chihoko** [comment]
@ericsysmin I agree, I didn't mean that this was an issue that needed to be resolved in the module, just one that we faced.

**Comment by dj-wasabi** [comment]
I just pushed an release (v2.2.0) to the forge.
